File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/AuthorsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AuthorsRow } from "./AuthorsRow";

export interface AuthorsColumns {
    AuthorId: Column<AuthorsRow>;
    AuthorName: Column<AuthorsRow>;
    Sex: Column<AuthorsRow>;
    Email: Column<AuthorsRow>;
    AuthorImage: Column<AuthorsRow>;
    CreateTime: Column<AuthorsRow>;
    UpdateTime: Column<AuthorsRow>;
}

export class AuthorsColumns extends ColumnsBase<AuthorsRow> {
    static readonly columnsKey = 'BookManage.Authors';
    static readonly Fields = fieldsProxy<AuthorsColumns>();
}

File: LMIS/LMIS.Web/Modules/BookManage/Authors/AuthorsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Authors")]
[BasedOnRow(typeof(AuthorsRow), CheckNames = true)]
public class AuthorsColumns
{
    [EditLink, AlignRight]
    public long AuthorId { get; set; }
    [EditLink]
    public string AuthorName { get; set; }
    public short Sex { get; set; }
    public string Email { get; set; }
    public string AuthorImage { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}